x-pack/plugins/observability_solution/observability_ai_assistant_app/public/plugin.tsx

```diff
@@ -111,6 +111,7 @@ export class ObservabilityAIAssistantAppPlugin
               appService={appService}
               coreStart={coreStart}
               pluginsStart={pluginsStart}
+              isServerless={!!pluginsStart.serverless}
```

Comment:
no need to depend on the serverless plugin to perform this check. The constructor of the plugin has the `buildFlavor`:

```TS
  constructor(initContext: PluginInitializerContext) {
    const isServerless = initContext.env.packageInfo.buildFlavor === 'serverless';
  }
```

Reply:
Thank you @afharo